optics-core/src/Optics/Internal/Profunctor.hs

```diff
@@ -472,20 +498,36 @@ instance Traversing IxFunArrow where
 ----------------------------------------
 
 class Traversing p => Mapping p where
-  roam :: ((a -> b) -> s -> t) -> p i a b -> p i s t
-  iroam :: ((i -> a -> b) -> s -> t) -> p j a b -> p (i -> j) s t
+  roam
+    :: ((a -> b) -> s -> t)
+    -> p i a b
+    -> p i s t
+
+  iroam
+    :: ((i -> a -> b) -> s -> t)
+    -> p       j  a b
+    -> p (i -> j) s t
+
+instance Mapping (Star ((,) Unit')) where
+  roam  f (Star k) = Star $ wrapUnit' . f (unwrapUnit' . k)
+  iroam f (Star k) = Star $ wrapUnit' . f (\_ -> unwrapUnit' . k)
+  {-# INLINE roam #-}
+  {-# INLINE iroam #-}
 
 instance Mapping FunArrow where
-  roam f (FunArrow k) = FunArrow (f k)
+  roam  f (FunArrow k) = FunArrow $ f k
+  iroam f (FunArrow k) = FunArrow $ f (const k)
   {-# INLINE roam #-}
+  {-# INLINE iroam #-}
 
-  iroam f (FunArrow k) = FunArrow (f (const k))
+instance Mapping (IxStar ((,) Unit')) where
```

Comment:
This is ugly. I'd guess, that `StrictFunArrow` is not much more code.

Reply:
Unfortunately not, you need to write instances for all profunctor classes (times two, because there is also StrictIxFunArrow).

We get these for free with `(Ix)Star` because we have instances for `Applicative f => (Ix)Star f` below `Mapping`.

Reply:
Alternative is Star StrictIdentity, but that type has unlawful Applicative instance (see https://github.com/ekmett/lens/issues/664#issuecomment-234434845). I don't know if that matters though.

Reply:
I'm not even sure if it's possible to do this with FunArrow.

`StrictFunArrow` is equivalent to `Star StrictIdentity`, so in pseudo-Haskell:

```haskell
newtype StrictFunArrow i a b = StrictFunArrow { runStrictFunArrow :: a -> !b }
```

when you have `StrictFunArrow k`, you need to force the result of application of `k`. But how can you do it when writing e.g. the definition of `roam`? `k` is just passed to `f` there, you can't influence what happens with the result of `k` applied to the value we're modifying.

Reply:
So the point is to `seq` only the fields focused by optic. `StrictIdentity` would force whole structure, won't it? Am I overthinking?

This is highly non-trivial, so proper `[Note]` should be in place.

Reply:
Yes, I now see that we need a token, forcing which will force proper parts.

But it's something one need to think about when seen first time, it looks too complicated. Don't make adam or andres or possible contributors to wonder what happens, but write a note. This is tricky, and should be clarified.

Reply:
> So the point is to `seq` only the fields focused by optic. `StrictIdentity` would force whole structure, won't it? Am I overthinking?

It would force everything that's on the rhs of `<*>` in the definition of traversals. Exactly which parts would be forced depends on the way the traversal is written, which is quite bad.

> This is highly non-trivial, so proper `[Note]` should be in place.

Yes, that indeed needs to be explained.

Reply:
I replaced `Unit'` with proper `Identity'` type so that its inner workings are less obscure and added explanation. It should be fairly clear now what's going on.